Terraform Azurerm CosmosDB Account Crash On Create

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_cosmosdb_account`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
variables information:
 cosmos_serverless  = false
cosmos_auto_scale_max_throughput  = 4000
variable "indexing_policy" {
  type = object({
    indexing_mode     = string
    included_paths    = list(string)
    excluded_paths    = list(string)
    composite_indexes = list(list(object({ path = string, order = string })))
  })
  default = null
}

variable "unique_key_paths" {
  type        = list(string)
  description = "A list of paths to use for the unique key."
  default     = null
}

locals {
  offer_type              = "Standard"
  kind                    = "GlobalDocumentDB"
  consistency_level       = "BoundedStaleness"
  max_interval_in_seconds = 5
  max_staleness_prefix    = 100
  capabilities            = var.cosmos_serverless ? [{ name = "EnableServerless" }] : []  
  autoscale_settings      = var.cosmos_serverless ? [] : [{ max_throughput = var.cosmos_auto_scale_max_throughput }]
  indexing_policies       = var.indexing_policy == null ? [] : [var.indexing_policy]
  is_keys_null            = var.unique_key_paths == null
  is_keys_empty           = var.unique_key_paths == null ? true : length(var.unique_key_paths) <= 0
  add_unique_keys         = local.is_keys_null || local.is_keys_empty ? [] : [var.unique_key_paths]
  cosmos_name             = "cosmos_name"
}

resource "azurerm_cosmosdb_account" "cosmos_account" {
  name                            = local.cosmos_name
  location                        = "East US 2"
  resource_group_name             =  "resource_group_name"
  offer_type                      = local.offer_type
  kind                            = local.kind
  enable_multiple_write_locations = false
  key_vault_key_id                = data.azurerm_key_vault_key.module.versionless_id

  tags = {}

  consistency_policy {
    consistency_level       = local.consistency_level
    max_interval_in_seconds = local.max_interval_in_seconds
    max_staleness_prefix    = local.max_staleness_prefix
  }

  enable_automatic_failover = false

  geo_location {
    location          = "East US 2"
    failover_priority = 0
  }

  dynamic "capabilities" {
    for_each = local.capabilities

    content {
      name = capabilities.value.name
    }
  }

  dynamic "geo_location" {
    for_each = []

    content {
      prefix            = "${local.cosmos_name}-${geo_location.value.location}"
      location          = geo_location.value.location
      failover_priority = geo_location.value.failover_priority
    }
  }

  is_virtual_network_filter_enabled = true
  public_network_access_enabled     = true
  ip_range_filter                   = "ips including microsoft portal with commas separating"

  virtual_network_rule {
    id = "subnet_id"
  }

  lifecycle {
    ignore_changes = [
      tags,
    ]
    prevent_destroy = true
  }
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->

### Panic Output

<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

azurerm_cosmosdb_account.cosmos_account: Still creating... [11m0s elapsed]
azurerm_cosmosdb_account.cosmos_account: Still creating... [11m10s elapsed]
azurerm_cosmosdb_account.cosmos_account: Still creating... [11m20s elapsed]
azurerm_cosmosdb_account.cosmos_account: Still creating... [11m30s elapsed]
azurerm_cosmosdb_account.cosmos_account: Still creating... [11m40s elapsed]
╷
│ Error: Plugin did not respond
│ 
│   on cosmosPersistence.tf line 27, in resource "azurerm_cosmosdb_account" "cosmos_account":
│   27: resource "azurerm_cosmosdb_account" "cosmos_account" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-azurerm_v2.99.0_x5.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x8 pc=0x4f9d151]

goroutine 244 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/cosmos.resourceCosmosDbAccountApiUpsert.func1()
	/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/cosmos/cosmosdb_account_resource.go:1204 +0x1b1
github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext.func1()
       /src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource/state.go:110 +0x21f
created by github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource.(*StateChangeConf).WaitForStateContext
	/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource/state.go:83 +0x1dd

Error: The terraform-provider-azurerm_v2.99.0_x5.exe plugin crashed!

### Expected Behaviour

Terraform to create azure cosmos database account

<!--- What should have happened? --->

### Actual Behaviour

<!--- What actually happened? --->

Crashed and gave panic message

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply` 

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

Nothing abnormal.  Deleted in Azure Portal before trying to recreate

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/internal/services/cosmos/cosmosdb_account_resource.go
+++ b/internal/services/cosmos/cosmosdb_account_resource.go
@@ -1162,7 +1162,7 @@ func resourceCosmosDbAccountApiUpsert(client *documentdb.DatabaseAccountsClient,
 			status := "Succeeded"
 			if props := resp.DatabaseAccountGetProperties; props != nil {
 
-				var locations  []documentdb.Location
+				var locations []documentdb.Location
 
 				if props.ReadLocations != nil {
 					locations = append(locations, *props.ReadLocations...)

--- a/internal/services/cosmos/cosmosdb_account_resource.go
+++ b/internal/services/cosmos/cosmosdb_account_resource.go
@@ -1161,7 +1161,16 @@ func resourceCosmosDbAccountApiUpsert(client *documentdb.DatabaseAccountsClient,
 			}
 			status := "Succeeded"
 			if props := resp.DatabaseAccountGetProperties; props != nil {
-				locations := append(*props.ReadLocations, *props.WriteLocations...)
+
+				var locations  []documentdb.Location
+
+				if props.ReadLocations != nil {
+					locations = append(locations, *props.ReadLocations...)
+				}
+				if props.WriteLocations != nil {
+					locations = append(locations, *props.WriteLocations...)
+				}
+
 				for _, l := range locations {
 					if status = *l.ProvisioningState; status == "Creating" || status == "Updating" || status == "Deleting" {
 						break // return the first non successful status.

